presto-common/src/main/java/com/facebook/presto/common/block/VariableWidthBlockBuilder.java

```diff
@@ -255,6 +259,8 @@ public BlockBuilder writeBytes(Slice source, int sourceIndex, int length)
     @Override
     public AbstractVariableWidthBlockBuilder writeBytes(byte[] source, int sourceIndex, int length)
     {
+        checkValidSliceRange(sourceIndex, length);
```

Comment:
I'm a little confused. Maybe the argument names are off, but if sourceIndex is an index into the source byte[] argument, which the name sounds like it is, then it shouldn't matter. What matters is where we are in the target slice.

sourceIndex could be quite large, e.g. a billion, but if it's only writing the next 10 bytes and we have room for that in sliceOutput, why do we care? Am I miusunderstanding how this works?

Reply:
The arguments are what is supposed to be. The `writeBytes` method transfers the bytes from the source buffer to the SliceOutput buffer. The validation was missing where if the source buffer is larger than the SliceOutput buffer and crossing the integer limit than it overflows and becomes negative. In either case it fails, but in the former case we throw IndexOutOfBoundException which we are fixing here by catching it early and throwing the right exception.

Reply:
OK, got it. I see what you're doing now. I suggest chanigng the method name to checkValidSourceRange or something like that, since the range bing checked is in the source, not the target

Reply:
This range we are checking for the Slice where we are writing to make sure it's within the range of destination slice range. So named it accordingly.